### ISSUE TITLE
Metabase integration: timestamp subtraction can’t be rendered (need to cast to text)

### ISSUE THREAD

**Issue by awang** [MaterializeInc/materialize#7861]
cc @mjibson

**Comment by mjibson** [comment]
I'm not sure what this means. How is this error produced? Does it show up if you just try to run metabase or is it due to a specific query?

**Comment by awang** [comment]
@mjibson Ah, so sorry about that!  I thought you had filed this issue (I thought it was related to #3727) but only now do I realize that it's completely unrelated.  

@antifuchs this is something you mentioned during your dogfood presentation -- do you think it's annoying enough to be a bugfix day candidate?

**Comment by antifuchs** [comment]
Ooh that would be interesting. I think it's definitely a common operation that users perform a bunch

**Comment by awang** [comment]
Thanks!  It's been awhile, but @antifuchs do you happen to remember what query caused this?

If it helps, the comment is on Day 1 ~57:00:   https://docs.google.com/document/d/1fdx4eIMraovmI4uJv0w-jVr4Y2CdFIzeQMI6ApDmH0A/edit

**Comment by antifuchs** [comment]
Sorry for misreading the question. I believe it was something like `SELECT NOW() - NOW()` in metabase:

This works:
![image](https://user-images.githubusercontent.com/11864/129958534-b721c730-401d-4001-b9f3-775b3d18f259.png)

This doesn't:
![image](https://user-images.githubusercontent.com/11864/129958581-c3dfe7cf-3503-46ae-a956-f33f330c81d8.png)

**Comment by mjibson** [comment]
Thanks for the info! That query causes the following to run:

```
SELECT
    typinput = 'array_in'::REGPROC AS is_array, typtype, typname
FROM
    pg_catalog.pg_type
    LEFT JOIN (
            SELECT
                ns.oid AS nspoid, ns.nspname, r.r
            FROM
                pg_namespace AS ns
                JOIN (
                        SELECT
                            s.r, (current_schemas(false))[s.r] AS nspname
                        FROM
                            ROWS FROM (
                                generate_series(1, array_upper(current_schemas(false), 1))
                            )
                                AS s (r)
                    )
                        AS r USING (nspname)
        )
            AS sp ON sp.nspoid = typnamespace
WHERE
    typname = $1
ORDER BY
    sp.r, pg_type.oid DESC;
```

Which errors with
```
ErrorResponse {
    severity: Error,
    code: SqlState(
        EXX000,
    ),
    message: "unknown catalog item 'regproc'",
    detail: None,
    hint: None,
    position: None,
}
```

**Comment by mjibson** [comment]
Also repros with `select '1s'::interval;`. I'm really curious what's going on.

**Comment by benesch** [comment]
Original issue was #7188. Seems like supporting regproc (#7790) has a very good shot at us just supporting this query?

**Comment by antifuchs** [comment]
Hah, I'd totally forgotten that original report. Yep, regproc is exactly what this is about.

**Comment by mjibson** [comment]
Closed by #8012
